common/utils/utils.go

```diff
@@ -71,10 +70,16 @@ func ReadGenesis(genesisPath string) (*core.Genesis, error) {
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if closeErr := file.Close(); closeErr != nil && err == nil {
+			err = closeErr
+		}
+	}()
 
 	genesis := new(core.Genesis)
 	if err := json.NewDecoder(file).Decode(genesis); err != nil {
-		return nil, errors.Join(err, file.Close())
+		return nil, err
 	}
-	return genesis, file.Close()
+	return genesis, nil
```

Comment:
No error will be reported to the caller if deferred close fails when returning from this line

Reply:
@HAOYUatHZ can you help to address this comment and merge?
```suggestion
	return genesis, err
```

(optional) Actually you may want to have an additional variable `retErr` to distinguish from widely used local variable `err`.